components/kyma-environment-broker/internal/component_input_helpers.go

```diff
@@ -0,0 +1,120 @@
+package internal
+
+import (
+	"context"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/ptr"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/reconciler"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	SCMigrationComponentName          = "sc-migration"
+	BTPOperatorComponentName          = "btp-operator"
+	HelmBrokerComponentName           = "helm-broker"
+	ServiceCatalogComponentName       = "service-catalog"
+	ServiceCatalogAddonsComponentName = "service-catalog-addons"
+	ServiceManagerComponentName       = "service-manager-proxy"
+)
+
+type ClusterIDGetter func() (string, error)
+
+func DisableServiceManagementComponents(r ProvisionerInputCreator) {
+	r.DisableOptionalComponent(SCMigrationComponentName)
+	r.DisableOptionalComponent(HelmBrokerComponentName)
+	r.DisableOptionalComponent(ServiceCatalogComponentName)
+	r.DisableOptionalComponent(ServiceCatalogAddonsComponentName)
+	r.DisableOptionalComponent(ServiceManagerComponentName)
+	r.DisableOptionalComponent(BTPOperatorComponentName)
+}
+
+func getBTPOperatorProvisioningOverrides(creds *ServiceManagerOperatorCredentials) []*gqlschema.ConfigEntryInput {
```

Comment:
this might be controversial, I would be interested in your opinion @piotrmiskiewicz and @ksputo, I very much disliked to write the same code for provisioning, updating and upgrade queue, I would like to propose this to serve as a common place and steps in the queues can reuse the same implementation.

Reply:
the plan was to have common implementation (to be reused) but there was no time for refactoring